src/rust/engine/src/externs.rs

```diff
@@ -101,6 +105,12 @@ impl Externs {
   pub fn create_exception(&self, msg: String) -> Value {
     (self.create_exception)(self.context, msg.as_ptr(), msg.len() as u64)
   }
+
+  pub fn invoke_runnable(&self, runnable: &Runnable) -> RunnableComplete {
+    let args_clone: Vec<Value> = runnable.args().clone();
```

Comment:
Is it necessary to clone here? I remember needing to in another case, but it seems like it should be fine not to here, and to just call `as_ptr` on the input (we know that the function we're calling won't alias it). If it's forcing you to, then nevermind.